### PR TITLE
feat: add SignUp component

### PR DESCRIPTION
**Pull request by D-Pagey**
_**This is work in progress**_ - Just making this Draft PR to compare to your branch

[Trello - Build Sign Up](https://trello.com/c/pSNP5V8H)

**Questions:**

- How should we handle shared components like buttons, inputs etc? I would guess an `inputs` folder in the main `components` folder?

### CODE REVIEW

File: src/components/SignUpPage/styles.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    flex-direction: column;
+    width: 300px;
+`;
+
+export const Label = styled.label`
+    align-items: center;
+    display: flex;
+    font-weight: ${fontWeight.medium};
+    justify-content: space-between;
+    margin: 5px 0;
```
Comment (navgarcha) [inline src/components/SignUpPage/styles.js:25]: use spacing token

File: src/components/Navbar/index.js [hunk 2]
```diff
@@ -19,6 +19,10 @@ const Navbar = () => {
                 <S.NavbarLink to="/stats">Stats</S.NavbarLink>
             </S.ListItem>
 
+            <S.ListItem>
+                <S.NavbarLink to="/signup">Sign Up</S.NavbarLink>
+            </S.ListItem>
```
Comment (navgarcha) [inline src/components/Navbar/index.js:24]: i think this should be added to the homepage and the login page but not the main navigation. if we do decide to keep it here it should disappear when we have an `authSession`

File: src/components/AuthProvider/index.js [hunk 3]
```diff
@@ -31,11 +31,27 @@ function AuthProvider({ children }) {
             .catch((err) => console.log('authLogout err', err)); // eslint-disable-line
     };
 
+    const authSignUp = async ({ email, password }) => {
+        Auth.signUp({
+            username: email,
+            password,
+            attributes: {
+                email
+            }
```
Comment (navgarcha) [inline src/components/AuthProvider/index.js:40]: what does this do?

**Comment by D-Pagey** [comment]
RIP

**Pull request closed without merging**